Full moderation tutorial with clean & infractions.md (#28)
MoonlightCapital/MoonlightBot-docs: moderation-commands/clean.md

# clean

> Delete a large amount of messages within the parameters given

Reviewer: Too brief

> Delete a large amount of messages. Determine which messages are deleted using parameters. Parameters are options that distinguish messages such as those that contain a specific phrase. Moderators can delete at most 100 messages within the last 14 days.

Reviewer: ```suggestion
This command deletes a large amount of messages from a channel. Determine which messages are deleted using parameters. Parameters are options that distinguish messages such as those that contain a specific phrase. Up to 100 messages can be deleted in a single execution of the command, and only those sent within the last 14 days.
```

i'd also replace the term "parameter" with "criteria", explain that multiple criteria will narrow down the deletion to only messages that fit all of them

> This command deletes a large amount of messages from a channel. Determine which messages are deleted using criteria. Criteria are options that distinguish messages such as those that contain a specific phrase. Using multiple criteria will narrow the selection to only messages that fit all of them. Up to 100 messages can be deleted in a single execution of the command, and only those sent within the last 14 days.

Reviewer: ```suggestion
This command deletes a large amount of messages from a channel, optionally limited by specifying some deletion criteria. Criteria are options that distinguish messages such as those that contain a specific phrase. Using multiple criteria will narrow the selection to only messages that fit all of them. Up to 100 messages can be deleted in a single execution of the command, and only those sent within the last 14 days.
```

### Options

* `amount`: The amount of messages you want to clean (The default is 100)
* `user`: The user you want to do delete the messages of
* `starts-with`: Delete messages that start with this value
* `ends-with`: Delete messages that end with this value
* `contains`: Delete messages that contain this value

> * `bots`: Whether or not to delete messages sent by bots (The default is false)

Reviewer: this works a bit differently whether it's explicitly false or "not set", refer to https://canary.discord.com/channels/359057740200673280/902924561073446953/1254732734169874523

* `system`: Whether or not to delete system messages. System messages are messages sent by Discord, such as join or pin messages

> ![An example of a system message](/.gitbook/assets/SystemMessage.png "System Message Example")

Reviewer: this image is unnecessary, remove

* `preserve-pins`: If true, pinned messages will not be deleted
* `no-trace`: If true, the confirmation message will only be visible to you

## Logs

> * `CHANNEL_CLEAN`: MoonlightBot will send two things: an HTML message that, when opened, displays the deleted messages, and a message detailing where messages were cleaned, how many were cleaned, who cleaned them, and the filters applied

Reviewer: inaccurate, too many commas

![An example of logs sent by the bot after a clean command execution](/.gitbook/assets/CleanLogs.png "Clean Logs")

{% hint style="info" %}

> To open an HTML file, download it and open the file explorer. Double-click the file, and your computer will open a new tab in your browser to display the executed code

Reviewer: this doesn't execute anything html is just to have a fancy visualization of the message, needs rewording

### Options

* `amount`: The amount of messages you want to clean (The default/maximum amount is 100)
* `user`: The user you want to do delete the messages of
* `starts-with`: Delete messages that start with this value
* `ends-with`: Delete messages that end with this value
* `contains`: Delete messages that contain this value

> * `bots`: Whether or not to delete messages sent by bots. If this option is left blank, the bot will delete these messages, but if left as true, the bot will only delete messages by bots, otherwise it will only delete messages not sent by bots

Reviewer: left as true? this sentence is generally very convoluted and needs rewording

and the system options works the same way

### Options

> * `amount`: The amount of messages you want to clean (The default/maximum amount is 100 messages)

Reviewer: ```suggestion
* `amount`: The amount of messages you want to clean (The default/maximum amount is 100)
```

* `user`: The user you want to do delete the messages of
* `starts-with`: Delete messages that start with this value
* `ends-with`: Delete messages that end with this value
* `contains`: Delete messages that contain this value

> * `bots`: Whether or not to delete messages sent by bots

Reviewer: This still needs to include the same information I linked in the previous comment

* `system`: Whether or not to delete system messages. System messages are messages sent by Discord, such as join or pin messages
* `preserve-pins`: If true, pinned messages will not be deleted
* `no-trace`: If true, the confirmation message will only be visible to you

## Logs

> * `CHANNEL_CLEAN`: MoonlightBot will send a message with two things: an HTML file that displays the deleted messages when opened and an embed detailing where messages were cleaned, how many were cleaned, who cleaned them, and the filters applied

Reviewer: ```suggestion
* `CHANNEL_CLEAN`: MoonlightBot will send a message with two things: an HTML file that contains the history of the deleted messages and an embed detailing where messages were cleaned, how many were cleaned, who cleaned them, and the filters applied
```
